pkg/manifestclient/readwrite_roundtripper.go

```diff
@@ -0,0 +1,94 @@
+package manifestclient
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// Enter here and call `NewForConfigAndClient(&rest.Config{}, httpClient)`
+func NewHTTPClient(mustGatherDir string) MutationTrackingClient {
+	mutationTrackingRoundTripper := newReadWriteRoundTripper(newMustGatherReader(mustGatherDir))
+	return &mutationTrackingClient{
+		httpClient: &http.Client{
+			Transport: mutationTrackingRoundTripper,
+		},
+		mutationTrackingRoundTripper: mutationTrackingRoundTripper,
+	}
+}
+
+// Enter here and call `NewForConfigAndClient(&rest.Config{}, httpClient)`
+func NewTestingHTTPClient(embedFS embed.FS, prefix string) MutationTrackingClient {
+	mutationTrackingRoundTripper := newReadWriteRoundTripper(newPrefixedReader(embedFS, prefix))
+	return &mutationTrackingClient{
+		httpClient: &http.Client{
+			Transport: mutationTrackingRoundTripper,
+		},
+		mutationTrackingRoundTripper: mutationTrackingRoundTripper,
+	}
+}
+
+func NewTestingRoundTripper(embedFS embed.FS, prefix string) *readWriteRoundTripper {
+	return newReadWriteRoundTripper(newPrefixedReader(embedFS, prefix))
+}
+
+func NewRoundTripper(mustGatherDir string) *readWriteRoundTripper {
+	return newReadWriteRoundTripper(newMustGatherReader(mustGatherDir))
+}
+
+func newReadWriteRoundTripper(contentReader RawReader) *readWriteRoundTripper {
+	return &readWriteRoundTripper{
+		readDelegate:  newReadRoundTripper(contentReader),
+		writeDelegate: newWriteRoundTripper(),
+	}
+}
+
+type readWriteRoundTripper struct {
+	readDelegate  *manifestRoundTripper
+	writeDelegate *writeTrackingRoundTripper
+}
+
+type MutationTrackingRoundTripper interface {
+	http.RoundTripper
+	GetMutations() *AllActionsTracker
+}
+
+type mutationTrackingClient struct {
+	httpClient *http.Client
+
+	mutationTrackingRoundTripper MutationTrackingRoundTripper
+}
+
+func (m mutationTrackingClient) GetHTTPClient() *http.Client {
+	return m.httpClient
+}
+
+func (m mutationTrackingClient) GetMutations() *AllActionsTracker {
+	return m.mutationTrackingRoundTripper.GetMutations()
+}
+
+type MutationTrackingClient interface {
+	GetHTTPClient() *http.Client
+	GetMutations() *AllActionsTracker
+}
+
+func (rt *readWriteRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	switch req.Method {
+	case "GET", "HEAD":
+		return rt.readDelegate.RoundTrip(req)
```

Comment:
have you pushed all your changes? i cannot find the readDelegate impl.

Reply:
> have you pushed all your changes? i cannot find the readDelegate impl.

It merged a couple months back.